searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java

```diff
@@ -0,0 +1,138 @@
+
+     */
+    public FaceDetect (String file, String type) {
+        Logger log = At18Logger.getLogger();
+        faceDetection (file, type);
+
+    }
+    /**
+
+     * Method that it's the process of the face detection where create the square in the face if it's detected
+
+     */
+
+    public static void faceDetection (String file, String type) {
+        Logger log = At18Logger.getLogger();
+        //Current_dir will take the direction of the path
+        current_dir = System.getProperty("user.dir");
+
+        //run the dll file, so the Open CV works correctly
+        System.load(current_dir + "\\src\\main\\resources\\libraries\\opencv\\x64\\opencv_java454.dll");
+
+        //imgFile it's the file we want to analyze if there it's a person in the photo
+        String imgFile = current_dir +"\\"+ file;
+
+        //Decodes the images
+        Mat src = Imgcodecs.imread(imgFile.toString());
+
+        // Are the connecting dots of a facial layout. This xml exactly defines a front-face layout
+        String xmlFile ="D:\\workspacejala\\progra102\\JU-OBJECT-RECOGNITION\\searchobject\\src\\main\\resources\\xml\\lbpcascade_frontalface.xml";
+
+        //cascade classifiers is an effective object detection method
+        CascadeClassifier cc = new CascadeClassifier(xmlFile);
+
+        //detectMultiScale used to detect the faces. This function will return a rectangle with coordinates(x,y,w,h)
+        MatOfRect faceDetection = new MatOfRect();
+
+        //faceDetection will determine the amount of face that has in the image
+        cc.detectMultiScale(src, faceDetection);
+
+        //Condition to detect if the input image its a profile or its more people
+        if (type.equals( "profile") && faceDetection.toArray().length == 1 ) {
+            for(Rect rect: faceDetection.toArray()) {
+                Imgproc.rectangle(src, new Point(rect.x, rect.y),
+                        new Point(rect.x + rect.width, rect.y + rect.height) , new Scalar(0, 0, 255), 3);
+            }
+
+            String finalfile = file.substring(22,file.length());
+
+            //Save the image already processed with the face detection
+            Imgcodecs.imwrite("D:\\workspacejala\\progra102\\JU-OBJECT-RECOGNITION\\searchobject\\src\\main\\resources\\images\\" + finalfile, src);
+
+            //logger: info that everything it right
+            log.info("It's only one person in the image");
+
+            //result if the image it's a profile
+            result = "profile detected";//"";
+
+        }  else if (type.equals("multiple") && faceDetection.toArray().length > 1 ) { //
+            for(Rect rect: faceDetection.toArray()) {
+                Imgproc.rectangle(src, new Point(rect.x, rect.y),
+                        new Point(rect.x + rect.width, rect.y + rect.height) , new Scalar(0, 0, 255), 3);
+            }
+            String finalfile = file.substring(22,file.length());
```

Comment:
the naming conventions, can be `finalFile`

Reply:
already change it